src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp

```diff
@@ -69,43 +70,43 @@ class boss_bronjahm : public CreatureScript
             boss_bronjahmAI(Creature* creature) : BossAI(creature, DATA_BRONJAHM)
             {
                 DoCast(me, SPELL_SOULSTORM_CHANNEL, true);
+                _soulFragmentsSpawned = 0;
             }
 
             void Reset() override
             {
+                _Reset();
                 events.Reset();
                 events.SetPhase(PHASE_1);
                 events.ScheduleEvent(EVENT_SHADOW_BOLT, 2000);
-                events.ScheduleEvent(EVENT_MAGIC_BANE, urand(8000, 20000));
+                events.ScheduleEvent(EVENT_MAGIC_BANE, urand(8000, 20000), 0, PHASE_1);
                 events.ScheduleEvent(EVENT_CORRUPT_SOUL, urand(25000, 35000), 0, PHASE_1);
-
-                instance->SetBossState(DATA_BRONJAHM, NOT_STARTED);
+                _soulFragmentsSpawned = 0;
             }
 
-           void JustReachedHome() override
-           {
-               DoCast(me, SPELL_SOULSTORM_CHANNEL, true);
-           }
-
             void EnterCombat(Unit* /*who*/) override
             {
-                Talk(SAY_AGGRO);
+                _EnterCombat();
                 me->RemoveAurasDueToSpell(SPELL_SOULSTORM_CHANNEL);
-
-                instance->SetBossState(DATA_BRONJAHM, IN_PROGRESS);
+                Talk(SAY_AGGRO);
             }
 
             void JustDied(Unit* /*killer*/) override
             {
-                Talk(SAY_DEATH);
+                _JustDied();
+                sCreatureTextMgr->SendSound(me, SOUND_DEATH, CHAT_MSG_MONSTER_YELL, 0, TEXT_RANGE_NORMAL, TEAM_OTHER, false);
+            }
```

Comment:
[**Nyeriah**](https://github.com/Nyeriah) added a note [19 hours ago](https://github.com/TrinityCore/TrinityCore/pull/14036#discussion_r24174111 "Feb. 5. 2015, 17:14 CET")
Use creature_text here (the entry is already in the database)